nilearn/decoding/decoder.py

```diff
@@ -870,6 +871,9 @@ def decision_function(self, X):
         y_pred: ndarray, shape (n_samples,)
             Predicted class label per sample.
         """
+        # for backwards compatibility
+        if not isinstance(X, type(np.array)):
+            X = self.masker_.transform(X)
```

Comment:
Sorry I though it was ready but then realized that this change to take X as a niimg was not backward compatible so I added this conditional. However, the docstring says X is a niimg like object so it can also be seen as a bug. Should we deprecate behavior here instead? Not sure how often people call this function directly but it is not even tested directly in our test suite